torch/distributed/algorithms/ddp_comm_hooks/optimizer_overlap_hooks.py

```diff
@@ -32,6 +32,32 @@ def _check_valid_functional_optim(self):
             )
 
 
+def _apply_optim_in_backward_hook(
```

Comment:
Should we rename this to suggest that it is a factory for the hook?

Otherwise, the only difference between this method and the actual hook defined inside is the preceding `_`, which may be confusing since the two have different quite semantics (one being a factory and one being the hook to run itself).